crates/solvers/src/infra/dex/zeroex/dto.rs

```diff
@@ -0,0 +1,160 @@
+        util::serialize,
+    },
+    bigdecimal::BigDecimal,
+    ethereum_types::{H160, U256},
+    serde::{Deserialize, Serialize},
+    serde_with::serde_as,
+};
+
+/// A 0x API quote query parameters.
+///
+/// See [API](https://docs.0x.org/0x-api-swap/api-references/get-swap-v1-quote)
+/// documentation for more detailed information on each parameter.
+#[serde_as]
+#[derive(Clone, Default, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Query {
+    /// Contract address of a token to sell.
+    pub sell_token: H160,
+
+    /// Contract address of a token to buy.
+    pub buy_token: H160,
+
+    /// Amount of a token to sell, set in atoms.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    #[serde_as(as = "Option<serialize::U256>")]
+    pub sell_amount: Option<U256>,
+
+    /// Amount of a token to sell, set in atoms.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    #[serde_as(as = "Option<serialize::U256>")]
+    pub buy_amount: Option<U256>,
+
+    /// Limit of price slippage you are willing to accept.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub slippage_percentage: Option<Slippage>,
+
+    /// The target gas price for the swap transaction.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    #[serde_as(as = "Option<serialize::U256>")]
+    pub gas_price: Option<U256>,
+
+    /// List of sources to exclude.
+    #[serde(skip_serializing_if = "Vec::is_empty", with = "comma_separated")]
+    pub excluded_sources: Vec<String>,
+
+    /// The affiliate address to use for tracking and analytics purposes.
+    pub affiliate_address: H160,
+
+    /// Requests trade routes which aim to protect against high slippage and MEV
+    /// attacks.
+    pub enable_slippage_protection: bool,
+}
+
+/// A 0x slippage amount.
+#[derive(Clone, Debug, Serialize)]
+pub struct Slippage(BigDecimal);
+
+impl Query {
+    pub fn with_domain(
+        self,
+        order: &dex::Order,
+        slippage: &dex::Slippage,
+        gas_price: auction::GasPrice,
+    ) -> Self {
+        let (sell_amount, buy_amount) = match order.side {
+            order::Side::Buy => (None, Some(order.amount.get())),
+            order::Side::Sell => (Some(order.amount.get()), None),
+        };
+
+        Self {
+            sell_token: order.sell.0,
+            buy_token: order.buy.0,
+            sell_amount,
+            buy_amount,
+            // Note that the API calls this "slippagePercentage", but it is **not** a
+            // percentage but a factor.
+            slippage_percentage: Some(Slippage(slippage.as_factor().clone())),
+            gas_price: Some(gas_price.0 .0),
+            ..self
+        }
+    }
+}
+
+/// A Ox API quote response.
+#[serde_as]
+#[derive(Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Quote {
+    /// The address of the contract to call in order to execute the swap.
+    pub to: H160,
+
+    #[serde_as(as = "serialize::Hex")]
+    /// The swap calldata.
```

Comment:
To make it consistent with the other fields.
```suggestion
    /// The swap calldata.
    #[serde_as(as = "serialize::Hex")]
```